src/etc/inc/interfaces.inc

```diff
@@ -2573,6 +2573,14 @@ EOD;
 			$mpdconf .= <<<EOD
 	set pppoe iface {$port}
 
+EOD;
+		}
+
+		if (($type == "l2tp") && !empty($ppp['secret'])) {
+			$secret = base64_decode($ppp['secret']);
+			$mpdconf .= <<<EOD
+	set l2tp secret {$secret}
```

Comment:
That should probably be enclosed in double quotes and have quotes escaped inside, to be safe.

Reply:
Done